tests/test_files.py

```diff
@@ -10,68 +11,47 @@
                          new_output)
 
 
-async def test_hidden_files(jp_fetch, jp_serverapp, jp_root_dir):
-    not_hidden = [
-        u'å b',
-        u'å b/ç. d',
-    ]
-    hidden = [
-        u'.å b',
-        u'å b/.ç d',
-    ]
-    dirs = not_hidden + hidden
-
-    for d in dirs:
-        path = jp_root_dir / d.replace('/', os.sep)
-        path.mkdir(parents=True, exist_ok=True)
-        path.joinpath('foo').write_text('foo')
-        path.joinpath('.foo').write_text('.foo')
-
-    for d in not_hidden:
-        r = await jp_fetch(
-            'files', d, 'foo',
-            method='GET'
-        )
-        assert r.body.decode() == 'foo'
-
-        with pytest.raises(tornado.httpclient.HTTPClientError) as e:
-            await jp_fetch(
-                'files', d, '.foo',
-                method='GET'
-            )
-        assert expected_http_error(e, 404)
-
-    for d in hidden:
-        for foo in ('foo', '.foo'):
-            with pytest.raises(tornado.httpclient.HTTPClientError) as e:
-                await jp_fetch(
-                    'files', d, foo,
-                    method='GET'
-                )
-            assert expected_http_error(e, 404)
+@pytest.fixture(params=[
+    [False, ['å b']],
+    [False, ['å b', 'ç. d']],
+    [True, ['.å b']],
+    [True, ['å b', '.ç d']]
+])
+def maybe_hidden(request):
+    return request.param
+
+
+async def fetch_expect_200(jp_fetch, *path_parts):
+    r = await jp_fetch('files', *path_parts, method='GET')
+    assert (r.body.decode() == path_parts[-1]), (path_parts, r.body)
+
+
+async def fetch_expect_404(jp_fetch, *path_parts):
+    with pytest.raises(tornado.httpclient.HTTPClientError) as e:
+        r = await jp_fetch('files', *path_parts, method='GET')
```

Comment:
This is a nice update - thank you.

Would you mind removing the unused variable?

Reply:
fixed in da0fef56e